jobs/payment-jobs/tests/jobs/test_routing_slip_task.py

```diff
@@ -334,17 +334,127 @@ def test_receipt_adjustments(session, rs_status):
     child_rs.save()
 
     parent_rs.status = rs_status
+    parent_rs.save()
 
     # Test exception path first.
-    with patch("pay_api.services.CFSService.adjust_receipt_to_zero") as mock:
-        mock.side_effect = Exception("ERROR!")
+    with patch("pay_api.services.CFSService.get_receipt") as mock_get_receipt, \
+         patch("pay_api.services.CFSService.adjust_receipt_to_zero") as mock_adjust:
+        mock_get_receipt.return_value = {
+            'unapplied_amount': 0.0,
+            'receipt_amount': 10.0,
+            'invoices': []
+        }
+        mock_adjust.side_effect = Exception("ERROR!")
         RoutingSlipTask.adjust_routing_slips()
 
     parent_rs = RoutingSlipModel.find_by_number(parent_rs.number)
-    assert parent_rs.remaining_amount == 10
+    assert parent_rs.remaining_amount == 20
+    assert parent_rs.cas_mismatch is True
+
+    parent_rs.cas_mismatch = False
+    parent_rs.save()
 
-    with patch("pay_api.services.CFSService.adjust_receipt_to_zero"):
+    with patch("pay_api.services.CFSService.get_receipt") as mock_get_receipt, \
+         patch("pay_api.services.CFSService.adjust_receipt_to_zero"):
+        mock_get_receipt.return_value = {
+            'unapplied_amount': 10.0,
+            'receipt_amount': 10.0,
+            'invoices': []
+        }
         RoutingSlipTask.adjust_routing_slips()
 
     parent_rs = RoutingSlipModel.find_by_number(parent_rs.number)
     assert parent_rs.remaining_amount == 0
+    assert parent_rs.cas_mismatch is not True
+
+
+def test_receipt_adjustments_amount_mismatch(session):
+    """Test routing slip adjustment fails when CFS amount doesn't match."""
+    rs_number = "12346"
+    factory_routing_slip_account(
+        number=rs_number,
+        status=CfsAccountStatus.ACTIVE.value,
+        total=100,
+        remaining_amount=50,
+    )
```

Comment:
Might need a multiple invoice test where `cas_mismatch` is False 
looks like yours test don't have it? 

EG multiple of these
```
 "invoices": [
        {
            "invoice_number": "REG01036828",
            "total": 15,
            "amount_applied": 15,
            "links": [
                {
                    "rel": "self",
                    "href": "https://xxx/invs/REG01036828/"
                }
            ]
        }
    ],
```